test/Dapper.AOT.Test/Verifiers/SqlDetection.cs

```diff
@@ -23,6 +23,24 @@ public void Foo(DbConnection conn)
         """, DefaultConfig, [Diagnostic(DapperAnalyzer.Diagnostics.ParseError)
     .WithLocation(0).WithArguments(46010, "Incorrect syntax near 111.")]);
 
+    [Fact]
+    public Task CSVerifyQuestionMarkInQuery_LikePseudoPositional() => CSVerifyAsync("""
+        using Dapper;
+        using System.Data.Common;
+    
+        [DapperAot(true)]
+        class SomeCode
+        {
+            public void Foo(DbConnection conn)
+            {
+                _ = conn.Query<int>("select {|#0:'this ?looks? like pseudo-positional'|}");
+                _ = conn.Query<int>("select 'this ?' + 'does not look like ? pseudo-positional' + 'because only 1 question mark is in every string part ?'");
```

Comment:
Needs a third test IMO: genuine pseudo-positional usage should not raise an error (unless it currently does because Dapper.AOT doesn't understand it, in which case: just let me know)

Reply:
Indeed it raises an error!

this usage 
```c#
_ = conn.Query<int>("select * from Orders where Id = ?id?", new Poco { Id = "1" });
```
reports:
```
    // /0/Test0.cs(11,66): error DAP206: SQL parse error 46010: 'Incorrect syntax near ?.'
VerifyCS.Diagnostic(DapperAnalyzer.DAP206).WithSpan(11, 66, 11, 66).WithArguments("46010", "Incorrect syntax near ?."),
```

I took a glance over original Dapper but did not find an SQL processor there. Let me know on how we can proceed here :)

Reply:
the ideas:
1) remember location of pseudo-parameter usage
2) discard the SQL parse error for correct pseudo-parameter usage

Reply:
discussed in call